src/MineCase.Protocol/Formats/Chat.cs

```
@@ -0,0 +1,382 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Chat"/> class.
+        /// </summary>
+        public Chat()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Chat"/> class with the specified component.
+        /// </summary>
+        /// <param name="component">The top-level component of this object.</param>
+        public Chat(ChatComponent component)
+        {
+            Component = component;
+        }
+
+        /// <summary>
+        /// Parses Chat from a JSON string.
+        /// </summary>
+        /// <param name="json">The JSON string</param>
+        /// <returns>A Chat object</returns>
+        public static Chat Parse(string json)
+        {
+            if (string.IsNullOrEmpty(json))
+                return null;
+
+            var jsonObject = JObject.Parse(json);
+            Handle(jsonObject);
+            json = jsonObject.ToString();
+            ChatComponent component = null;
+
+            if (json.Contains("text"))
+            {
+                component = JsonConvert.DeserializeObject<StringComponent>(json);
+            }
+            else if (json.Contains("translate"))
+            {
+                component = JsonConvert.DeserializeObject<TranslationComponent>(json);
+            }
+            else if (json.Contains("keybind"))
+            {
+                component = JsonConvert.DeserializeObject<KeybindComponent>(json);
+            }
+            else if (json.Contains("score"))
+            {
+                component = JsonConvert.DeserializeObject<ScoreComponent>(json);
+            }
+            else if (json.Contains("selector"))
+            {
+                component = JsonConvert.DeserializeObject<SelectorComponent>(json);
+            }
+
+            if (component != null)
+            {
+                return new Chat(component);
+            }
+
+            return null;
```

Should throw an exception.